_posts/2019-09-26-2019-building-admissions.md

```diff
@@ -0,0 +1,284 @@
+3. Using the result of the API search, process the user's result
+   1. In the event a user **is** a contributor, have them confirm the email address they want to use for Slack, use the Slack API to send an invite, and finally congratulate them.
+   2. If they **have not** contributed we need to notify them of their ineligibility
+
+> Note: For our application we've renamed `PageController` to `EligibilityController` to fit with our school admissions theme.
+
+### Login with GitHub
+
+Starting from a new Phoenix project (`mix phx.new admissions`)  we looked at how to support GitHub login. For that we need a new dependency: `ueberauth_github`:
+
+```elixir
+  defp deps do
+    [
+      {:gettext, "~> 0.11"},
+      {:phoenix, "~> 1.4.0"},
+      {:phoenix_html, "~> 2.11"},
+      {:plug_cowboy, "~> 2.0"},
+      {:ueberauth_github, "~> 0.7.0"},
+
+      {:phoenix_live_reload, "~> 1.2", only: :dev}
+    ]
+  end
+```
+
+> We won't need to include `ueberauth` itself, as a dependency  of `ueberauth_github` it is included for us.
+
+> Helpful tip: Did you know you can use `mix hex.info <package name>` to get the latest version? Try it!
+
+With our application empowered with our new dependency what's left to do? Plenty! To finish our integration with Ueberauth we had a few subtasks:
+
+1. Create a `AuthController` that'll handle the callback phase of the OAuth request.
+
+2. Include our new controller and route in our `router.ex` file.
+
+3. Put the required configuration for Ueberauth in our `config/config.exs` file.
+
+4. Add a button to the UI for login. While we won't spend time in this article building the UI, we will touch on the required pieces.
+
+5. Setting up your application on GitHub. Here you'll also need to retrieve your `CLIENT_ID` and `CLIENT_SECRET` .
+
+   > GitHub setup and configuration goes beyond this article. If you aren't quite sure what to do, head over to GitHub's Developer article [Authorizing OAuth Apps](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/) 
+
+Onward!
+
+#### Our new controller
+
+Completing our first subtask requires we create a new controller for Ueberauth that will handle the OAuth callback from GitHub in the event of successful login. The only hard requirement for our controller is that we include the Ueberauth plug:
+
+```elixir
+defmodule AdmissionsWeb.AuthController do
+  use AdmissionsWeb, :controller
+
+  plug Ueberauth
+end
+```
+
+With the plug now in place we defined our the function to handle our requests which we've elected to name `callback/2`. This function  needs to retrieve the user details Ueberauth has so conveniently placed into the `Plug.Conn` assigns for us. The fields we're concerned with are the user's email and GitHub nickname:
```

Comment:
"With the plug now in place we defined our the function to handle our requests which we've elected to name `callback/2`" -> "With the plug in place, we'll define a function to handle our requests. We've elected to name that function `callback/2`".